user/src/main/java/com/chronica/user/logic/security/FilterBeforeRequest.java

```diff
@@ -0,0 +1,60 @@
+package com.chronica.user.logic.security;
+
+import com.chronica.user.data.constant.Role;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import com.chronica.user.logic.basic.AccountBasicService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class FilterBeforeRequest extends OncePerRequestFilter {
```

Comment:
Na polski - filtrujPrzedRequestem - czynność, czyli metoda. Odpowiednia nazwa to mogłoby być RequestAuthenticator